spec/inertia/partial_reload_spec.rb

```diff
@@ -0,0 +1,88 @@
+RSpec.describe 'partial reloads', type: :request do
+  describe 'optimization guard rails' do
+    context 'when a non-requested prop is defined as a value' do
+      let (:fake_std_err) {FakeStdErr.new}
+      let (:warn_message) {'[InertiaRails]: WARNING! The :expensive_prop prop is being computed even though your partial reload did not request it because it is defined as a value. You might want to wrap these in a callable like a lambda ->{} or InertiaRails::Lazy().'}
+      let (:warn_message_with_multiple) {'[InertiaRails]: WARNING! The :expensive_prop, :another_expensive_prop props are being computed even though your partial reload did not request them because they are defined as values. You might want to wrap these in a callable like a lambda ->{} or InertiaRails::Lazy().'}
+
+      around(:each) do |example|
+        begin
+          original_stderr = $stderr
```

Comment:
The complexity in this test is what tipped me off that I should be uncomfortable with the warning method above

Reply:
I think I could make this cleaner by warning via Rails.logger instead of going all the way down to Kernel.warn.

The important bit is above though, whether logging in dev/test is a useful interface for warning devs or if it's too easily ignored to be helpful.